apis/datadoghq/v2alpha1/test/builder.go

```diff
@@ -0,0 +1,273 @@
+	return builder
+}
+
+// Orchestrator Explorer
+
+func (builder *DatadogAgentBuilder) initOE() {
+	if builder.datadogAgent.Spec.Features.OrchestratorExplorer == nil {
+		builder.datadogAgent.Spec.Features.OrchestratorExplorer = &v2alpha1.OrchestratorExplorerFeatureConfig{}
+	}
+}
+
+func (builder *DatadogAgentBuilder) WithOrchestratorExplorerEnabled(enabled bool) *DatadogAgentBuilder {
+	builder.initOE()
+	builder.datadogAgent.Spec.Features.OrchestratorExplorer.Enabled = apiutils.NewBoolPointer(enabled)
+	return builder
+}
+
+func (builder *DatadogAgentBuilder) WithOrchestratorExplorerScrubContainersEnabled(enabled bool) *DatadogAgentBuilder {
+	builder.initOE()
+	builder.datadogAgent.Spec.Features.OrchestratorExplorer.ScrubContainers = apiutils.NewBoolPointer(enabled)
+	return builder
+}
+
+func (builder *DatadogAgentBuilder) WithOrchestratorExplorerExtraTags(tags []string) *DatadogAgentBuilder {
+	builder.initOE()
+	builder.datadogAgent.Spec.Features.OrchestratorExplorer.ExtraTags = tags
+	return builder
+}
+
+func (builder *DatadogAgentBuilder) WithOrchestratorExplorerDDUrl(ddUrl string) *DatadogAgentBuilder {
+	builder.initOE()
+	builder.datadogAgent.Spec.Features.OrchestratorExplorer.DDUrl = apiutils.NewStringPointer(ddUrl)
+	return builder
+}
+
+func (builder *DatadogAgentBuilder) WithOrchestratorExplorerCustomConfigData(customConfigData string) *DatadogAgentBuilder {
+	builder.initOE()
+	builder.datadogAgent.Spec.Features.OrchestratorExplorer.Conf = &v2alpha1.CustomConfig{
+		ConfigData: &customConfigData,
+	}
+	return builder
+}
+
+// Cluster Checks
+
+func (builder *DatadogAgentBuilder) initCC() {
+	if builder.datadogAgent.Spec.Features.ClusterChecks == nil {
+		builder.datadogAgent.Spec.Features.ClusterChecks = &v2alpha1.ClusterChecksFeatureConfig{}
+	}
+}
+
+func (builder *DatadogAgentBuilder) WithClusterChecksEnabled(enabled bool) *DatadogAgentBuilder {
+	builder.initCC()
+	builder.datadogAgent.Spec.Features.ClusterChecks.Enabled = apiutils.NewBoolPointer(enabled)
+	return builder
+}
+
+func (builder *DatadogAgentBuilder) WithClusterChecksUseCLCEnabled(enabled bool) *DatadogAgentBuilder {
+	builder.initCC()
+	builder.datadogAgent.Spec.Features.ClusterChecks.UseClusterChecksRunners = apiutils.NewBoolPointer(enabled)
+	return builder
+}
+
+// Prometheus Scrape
+
+func (builder *DatadogAgentBuilder) initPrometheusScrape() {
+	if builder.datadogAgent.Spec.Features.PrometheusScrape == nil {
+		builder.datadogAgent.Spec.Features.PrometheusScrape = &v2alpha1.PrometheusScrapeFeatureConfig{}
+	}
+}
+
+func (builder *DatadogAgentBuilder) WithPrometheusScrapeEnabled(enabled bool) *DatadogAgentBuilder {
+	builder.initPrometheusScrape()
+	builder.datadogAgent.Spec.Features.PrometheusScrape.Enabled = apiutils.NewBoolPointer(enabled)
+	return builder
+}
+
+func (builder *DatadogAgentBuilder) WithPrometheusScrapeServiceEndpointEnabled(enabled bool) *DatadogAgentBuilder {
```

Comment:
```suggestion
func (builder *DatadogAgentBuilder) WithPrometheusScrapeServiceEndpoints(enabled bool) *DatadogAgentBuilder {
```

(wish we had been consistent in having this option name as a description rather than an action -_-)